Support installing from git #1309

vellotis opened the pull request:
- Restore `postinstall #1305
- Move `babel-runtime` from `dependencies` to `devDependencies` as it is not required in production
- Add `buildDependencies` key to package.json that lists all devDependencies that are needed for building

Resolves #1309, #1067

tgriesser commented on package.json:
```diff
@@ -63,6 +64,14 @@
   },
+  "buildDependencies": [
+    "babel-cli",
+    "babel-runtime",
```
`babel-runtime` shouldn't actually be needed here since it's already in `dependencies`, right?

vellotis commented:
Actually yeah, it is redundant. I misunderstood the `babel-runtime` dependency and moved it under `devDependencies`. I understood my mistake and moved it back. But the value vas left in `buildDependencies`.

Will fix it.
